backend/src/main/java/com/bakdata/conquery/models/auth/develop/UserIdTokenExtractor.java

```diff
@@ -0,0 +1,56 @@
+package com.bakdata.conquery.models.auth.develop;
+
+import com.bakdata.conquery.models.auth.entities.User;
+import com.bakdata.conquery.models.auth.web.DefaultAuthFilter;
+import com.bakdata.conquery.models.identifiable.ids.specific.UserId;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.shiro.authc.AuthenticationToken;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.HttpHeaders;
+
+@RequiredArgsConstructor
+public class UserIdTokenExtractor implements DefaultAuthFilter.TokenExtractor {
+
+	private static final String UID_QUERY_STRING_PARAMETER = "access_token";
+
+	private final User defaultUser;
+
+	/**
+	 * Tries to extract a plain {@link UserId} from the request to submit it for the authentication process.
+	 */
+	@Override
+	public AuthenticationToken apply(ContainerRequestContext requestContext) {
+		// Check if the developer passed a UserId under whose the Request should be
+		// executed
+
+		// Check the Authorization header for a String which can be parsed as a UserId
+		String uid = requestContext.getHeaders().getFirst(HttpHeaders.AUTHORIZATION);
+
+		if (uid != null) {
```

Comment:
Would it make sense to remove the if case here and only keep the else case, as in: check for `if (uid == null)`?

Then you'd have to do the Bearer replace only once, some lines below, no?

Reply:
The second bearer replace seems obsolete ;) 
Thanks

Reply:
ist raus